editor/filesystem_dock.cpp

```diff
@@ -1599,6 +1599,21 @@ void FileSystemDock::_update_dependencies_after_move(const HashMap<String, Strin
 	}
 }
 
+void FileSystemDock::_update_folder_colors_after_move(const HashMap<String, String> &p_folders_renames) {
+	for (const KeyValue<String, String> &rename : p_folders_renames) {
+		if (assigned_folder_colors.has(rename.key)) {
+			assigned_folder_colors[rename.value] = assigned_folder_colors[rename.key];
+			assigned_folder_colors.erase(rename.key);
+		}
+	}
+
+	if (!ProjectSettings::get_singleton()->has_setting("file_customization/folder_colors")) {
```

Comment:
What is this for? Moving folders will only update existing paths.

Reply:
True, will update, realized it must have been assigned already or exist